cmd/gateway/azure/gateway-azure.go

```diff
@@ -696,34 +739,24 @@ func (a *azureObjects) GetObject(ctx context.Context, bucket, object string, sta
 		return azureToObjectError(minio.InvalidRange{}, bucket, object)
 	}
 
-	blobRange := &storage.BlobRange{Start: uint64(startOffset)}
-	if length > 0 {
-		blobRange.End = uint64(startOffset + length - 1)
-	}
-
-	blob := a.client.GetContainerReference(bucket).GetBlobReference(object)
-	var rc io.ReadCloser
-	var err error
-	if startOffset == 0 && length == 0 {
-		rc, err = blob.Get(nil)
-	} else {
-		rc, err = blob.GetRange(&storage.GetBlobRangeOptions{
-			Range: blobRange,
-		})
-	}
+	blobURL := a.client.NewContainerURL(bucket).NewBlobURL(object)
+	blob, err := blobURL.Download(ctx, startOffset, length, azblob.BlobAccessConditions{}, false)
 	if err != nil {
 		return azureToObjectError(err, bucket, object)
 	}
+
+	rc := blob.Body(azblob.RetryReaderOptions{})
```

Comment:
I am not able to download a file of 115 Mb with mc and with aws cli. This is an example of the aws error:

```
aws s3api --endpoint-url http://localhost:9000 get-object --bucket testbucket --key 115mb.bin /tmp/output.bin

4161536 read, but total bytes expected is 120586240.
```

The read bytes is random. But this issue doesn't happen with master version, I I am not sure what is the reason of that, should we configure some retries here ? Does the old API do that @c-w ?

Reply:
Good catch. Indeed, the old SDK by default retries downloads up to 5 times (see [defaultRetryAttempts](https://github.com/Azure/azure-sdk-for-go/blob/0820c0553d6c410603056869f7b0e52d61ce5bee/storage/client.go#L52)). I added retries to this PR in 7289ccb and verified by uploading/downloading a 350 MB file via the MinIO web dashboard and the AWS CLI.

Reply:
> Good catch. Indeed, the old SDK by default retries downloads up to 5 times (see [defaultRetryAttempts](https://github.com/Azure/azure-sdk-for-go/blob/0820c0553d6c410603056869f7b0e52d61ce5bee/storage/client.go#L52)). I added retries to this PR in [7289ccb](https://github.com/minio/minio/commit/7289ccb06225f44a901237723363684206c07e54) and verified by uploading/downloading a 350 MB file via the MinIO web dashboard and the AWS CLI.

Yes, it also works fine for me now, thanks!